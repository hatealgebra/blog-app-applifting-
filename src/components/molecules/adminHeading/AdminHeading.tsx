import React from 'react';
import Button from '../../atoms/button/Button';
import { ButtonLink } from '../../atoms/links/link.styled';
import { StyledMyArticlesHeading } from './adminHeading.styled';

const AdminHeading = ({
  heading,
  to,
  buttonText,
  isFormPage,
}: AdminHeadingProps) => {
  return (
    <StyledMyArticlesHeading>
      <h1>{heading}</h1>
      {to && (
        <ButtonLink colortheme="primary" to={to}>
          {buttonText}
        </ButtonLink>
      )}
      {isFormPage && <Button type="submit">{buttonText}</Button>}
    </StyledMyArticlesHeading>
  );
};

interface AdminHeadingProps {
  heading: string;
  buttonText: string;
  to?: string;
  isFormPage?: boolean;
}

export default AdminHeading;
