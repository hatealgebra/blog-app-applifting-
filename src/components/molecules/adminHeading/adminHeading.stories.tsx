import React from 'react';
import AdminHeading from './AdminHeading';

export const AdminHeadingExample = () => (
  <AdminHeading heading="Page Title" buttonText="Create new article" to="#" />
);

export default {
  title: 'Molecules/Admin Heading',
  component: AdminHeading,
};
