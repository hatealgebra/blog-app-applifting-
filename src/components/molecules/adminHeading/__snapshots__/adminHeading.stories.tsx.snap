// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Admin Heading AdminHeadingExample smoke-test 1`] = `
<div class="sc-hBpgZr blYtEN">
  <h1>
    Page Title
  </h1>
  <a colortheme="primary"
     class="sc-dkjKgF sc-csmVar iJlhGR blIAmy"
     href="/iframe.html/#"
  >
    Create new article
  </a>
</div>
`;
